### DISCUSSION TITLE
백엔드 만들기

### DISCUSSION THREAD

**Discussion by DongjaJ** [prgrms-fe-devcourse/FEDC4_TMI_HOMERS_OFF#358] [General]
4가지 후보들을 생각해봤는데 각자 하나씩 만들어보고 좋은 것으로 결정하는 것이 어떤가 합니다!

1. Firebase
2. Supabase
3. Strapi
4. sanity

**Comment by GBAJS754** [comment]
저는 1번이요!

**Comment by khakhiD** [comment]
Chat GPT 4o에게 장단점과 적합성을 비교해 달라고 했습니다.
러닝 커브가 있어 보이는 sanity는 제외했습니다.

**Firebase**
- 장점:
  - 실시간 데이터베이스: Firestore를 사용하면 실시간 데이터 동기화가 가능하며, 이는 댓글 및 게시글 업데이트에 유용합니다.
  - 인증 서비스: Firebase Authentication을 통해 간편하게 로그인, 로그아웃 및 사용자 관리를 할 수 있습니다.
  - 확장성: Google Cloud Platform과 통합되어 있어 필요에 따라 확장 가능합니다.
  - 호스팅 및 Functions: Firebase Hosting과 Cloud Functions를 사용해 서버리스 백엔드를 구축할 수 있습니다.
- 단점:
  - 복잡한 쿼리 제한: Firestore는 복잡한 쿼리에 제한이 있어 관계형 데이터베이스와 같은 유연성을 제공하지 않을 수 있습니다.
  - 벤더 락인: Firebase에 깊이 의존하게 되면 다른 플랫폼으로 이전이 어려울 수 있습니다.
- 적합성:
  - 실시간 데이터 동기화와 간편한 인증이 필요한 CRUD 게시판 프로젝트에 매우 적합합니다. 

**Supabase**
- 장점:
  - PostgreSQL 기반: 완전한 SQL 데이터베이스 기능을 제공하여 복잡한 쿼리 및 관계형 데이터베이스 구조를 쉽게 다룰 수 있습니다.
  - 실시간 기능: PostgreSQL을 기반으로 실시간 데이터 동기화를 지원합니다.
  - 인증 및 스토리지: 내장된 인증 서비스와 파일 스토리지를 제공합니다.
  - 오픈 소스: 오픈 소스로 제공되어 커스터마이징과 자체 호스팅이 가능합니다.
- 단점:
  - 새로운 플랫폼: Firebase만큼 성숙하지 않았으며, 커뮤니티와 에코시스템이 비교적 작습니다.
  - 성능 최적화 필요: PostgreSQL의 성능 최적화와 관련된 추가 작업이 필요할 수 있습니다.
- 적합성:
  - SQL 기반 데이터베이스와 복잡한 쿼리 지원이 필요한 경우 적합하며, 실시간 기능도 제공하므로 댓글 및 게시글 업데이트에 유용합니다. 

**Strapi**
- 장점:
  - 헤드리스 CMS: 콘텐츠 관리가 매우 유연하고 강력합니다.
  - RESTful 및 GraphQL API: 다양한 API 옵션을 제공하여 프론트엔드와 쉽게 통신할 수 있습니다.
  - 플러그인 시스템: 커뮤니티에서 만든 다양한 플러그인을 활용할 수 있습니다.
  - 자체 호스팅 가능: 호스팅 환경을 자유롭게 선택할 수 있습니다.
- 단점:
  - 설정 및 유지보수: 다른 BaaS (Backend as a Service) 옵션보다 설정과 유지보수가 더 필요합니다.
  - 실시간 데이터 지원 부족: 기본적으로 실시간 데이터를 지원하지 않지만, 추가 설정으로 구현할 수 있습니다.
- 적합성:
  - 콘텐츠 중심의 프로젝트에 적합하며, 다양한 API 통신 옵션과 커스터마이징이 필요한 경우 유용합니다. 그러나 실시간 기능은 추가 구현이 필요합니다.


Firebase가 압도적 > 그 다음이 Strapi > 그 다음이 supabase 순으로 한글 자료가 많은 것 같습니다.
저는 Firebase 혹은 Strapi 둘 중 하나로 선택지 좁히겠습니다.

**Reply by khakhiD** [reply]
최종 선택 Firebase
다른 의견도 충분히 환영. (사실 어떤 것을 써도 반대 없음)
